simplq/src/components/pages/Join/JoinForm.jsx

```diff
@@ -1,79 +1,167 @@
-import React, { useState, useRef, useEffect } from 'react';
-import { useSelector } from 'react-redux';
-
-import { handleEnterPress } from 'utils/eventHandling';
+import React, {useState, useRef, useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {handleEnterPress} from 'utils/eventHandling';
 import InputField from 'components/common/InputField';
 import PhoneInput from 'components/common/PhoneInput';
-import Button from 'components/common/Button';
+import StandardButton from 'components/common/Button';
 import LoadingStatus from 'components/common/Loading';
-import styles from './JoinPage.module.scss';
-
-export function JoinQueueForm({ joinQueueHandler, buttonText }) {
-  const [name, setName] = useState('');
-  const [invalidName, setInvalidName] = useState(false);
-  const [invalidContact, setInvalidContact] = useState(false);
-  const [contact, setContact] = useState('');
-  const actionStatus = useSelector((state) => state.actionStatus['joinQueue']);
-  const prevActionStatus = useRef();
-
-  useEffect(() => {
-    // Reset form only after successful action
-    if (prevActionStatus.current === 'pending' && actionStatus === 'fulfilled') {
-      setContact('');
-      setName('');
-    }
+import styles from './JoinForm.module.scss';
+import Step from "@material-ui/core/Step";
+import StepContent from "@material-ui/core/StepContent";
+import Box from "@material-ui/core/Box";
+import {Stepper} from "@material-ui/core";
+import StepLabel from "@material-ui/core/StepLabel";
+import Typography from "@material-ui/core/Typography";
+import {selectQueueInfo} from "../../../store/queueInfo";
+import {setErrorPopupMessage} from "../../../store/appSlice";
 
-    // Set previous action status for next render
-    prevActionStatus.current = actionStatus;
-  }, [actionStatus]);
+export function JoinQueueForm(props) {
+    const [name, setName] = useState('');
+    const [invalidName, setInvalidName] = useState(false);
+    const [invalidContact, setInvalidContact] = useState(false);
+    const [contact, setContact] = useState('');
+    const actionStatus = useSelector((state) => state.actionStatus['joinQueue']);
+    const prevActionStatus = useRef();
+    const [activeStep, setActiveStep] = React.useState(0);
+    const queueInfo = useSelector(selectQueueInfo);
+    const dispatch = useDispatch();
 
-  function handleNameChange(e) {
-    if (name.match('^[A-Za-z0-9 ]*$')) {
-      setName(e.target.value);
-      setInvalidName(false);
-    } else {
-      setInvalidName(true);
-    }
-  }
+    const handleNext = async () => {
+        if (invalidContact) return
+        if (contact === "") {
+            setInvalidContact(true)
+            return
+        }
+
+        if (props.queuePage) {
+            setActiveStep((prevActiveStep) => prevActiveStep + 1);
+            return
+        }
+
+        let result = await props.onSubmitGetToken(contact)
```

Comment:
I know it is not a best approach, I did that only to get result for checking if to display error message you provided, based on token available or not. So basically now there are 2 error messages, so possibly one can be removed and this could be refactored.

Reply:
Let's remove the error message we issue from here.